Rolling update

milljm opened the pull request.

Commit:
Update GCC and Clang to latest available

Attempt a GCC 8.2.0, and LLVM 7.0.1 build.
Remove CPPUNIT.
Fix miniconda/pkgs/urls.txt missing, causing permission errors when a user attempts to create an environment.
Fix chown issue after profile is created.
Remove make_payload.sh. That script no longer does anything as far as I can tell.

Closes #186 #188

Commit:
Appears lld does not yet work on Darwin platforms

Create symbolic link to lld only on linux machines for now.

Commit:
Support multiple shell environments

bash, sh, zsh, csh, tsch supported.

Do not modify user-owned environment (linux only). Instead, add
modules to the system-wide environment. Inform the user which
modules they should load. This will make new installations easier
when a user is unfamiliar with managing their own profiles.

Commit:
Update how we handle the code sign cert

Check for the presence of moose-codesign before attempting
to delete it.

For now, see if we even need the code-signing cert. This is
TEST! I suspect, the get-task-allow entittlement is all that
is required when using LLDB (any LLDB. even ours).

Commit:
Force loading of miniconda module

When using moose-dev-gcc, on a darwin machine, one must also
load the miniconda module due to python-config returning clang-
only cflags.

Closes #191

Commit:
re-enable moose-codesign cert

I neglected to realize codesigning is not intended for the apps being built
by our LLVM compiler, but for our LLVM debugger to actually operate on the
users machine.

Commit:
Deal with lsb_release 'No LSB modules'

We shouldn't consider stderr being a non-null value as an indication
as to a general non-zero exit code. Use poll instead.

Debian `stretch` has no solution for 'No LSB modules are available'.

Commit:
Automake patch which should fix the help2man

We are experiencing this bug on Ubuntu-18:
http://gnu-automake.7480.n7.nabble.com/bug-30631-Automake-1-6-fails-to-build-with-perl-5-18-2-td22865.html

1.16.1 contains the patch fix.

Commit:
Fix Debian stretch

Debian machines create a symlink from swig3 to swig. Do the same
in miniconda's swig so as to allow our swig to be used when on
this platform

Commit:
Revert to working GCC/LLVM

Do not load moose modules by default.

Allow moose-$$$ modules to be created based on dependencies.
This will allow a cleaner build when using --build-only options.

Better regex for finished dependencies. The regex routine was
not being specific enough.

Commit:
Further modularize the moose-tools module

Commit:
Add a patch routine for seacas module

A server hosting a meta package seacas depends on went down. This
change makes the installer point to mooseframework which now holds
the meta package in question.

Commit:
Install specific version of anytree

Commit:
The moose group says GCC 8 is working in moose

Bring gcc 8 and llvm 7 back. Creating a new branch
as there were other additional changes needed.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/300191/) on 315ea90 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/193/site)

This comment will be updated on new commits.

Commit:
Use LLVM 6.0.1

Revert to LLVM 6.0.1. We are unable to build PETSc 3.8.3 with latest LLVM.

Modify all civet-related petsc targets to load miniconda. Miniconda was
being loaded by moose-tools previously.

Correct the try_again function (convert it to an array). The function was
not detecting anything except the last command run.

Commit:
Add miniconda module

Found another module which needed to load miniconda along with moose-tools.

milljm commented:
Referencing #188

Commit:
fix sed -i creating backup modules

Fix mpich being built incorrectly. Closes #194
Force moose-tools to load when one loads moose-dev-$$$. Closes #191
Make the installation of the package create a global environment. Closes #190

Commit:
We should not use O3 optimizations with mpich

It would seem on Darwin, mpich 3.2 with enable-fast=O3 causes several tests to segfault.

Refrences #194

Commit:
Remove --enable-fast for mpich-clang

I am seeing errors during runtime with parallel tests. The backtraces point to
libpmpi.12.dylib. Much like this post:

   https://lists.mpich.org/pipermail/discuss/2016-July/004824.html

This PR is just a test to see if the newly added option (because I was doing it
wrong for the past 3 years), is the cause.

Added a message to be displayed about the users environment being changed (Macs).

Commit:
Upgrade to MPICH 3.3

Initial manual tests succeeded with mpich 3.3 w/ --enable-fast= w/clang

Commit:
Giving up on making mpich-3.3 work.

Setting enable-fast flag to O2 for mpich 3.2

Commit:
Just remove --enable-fast from clang mpich...

Remove the argument... see what happens

Commit:
Revert "Just remove --enable-fast from clang mpich..."

This reverts commit 85b8641d7058fa052bfb0305384834e5c9984155.

Commit:
Revert "Giving up on making mpich-3.3 work."

This reverts commit 4239c7f724f54bd2c6dda1577e44f9ad522294f1.
